charts/ccsm-helm/test/integration/integration_test.go

```diff
@@ -79,17 +85,143 @@ func (s *integrationTest) TestGatewayConnection() {
 	s.Require().NoError(err, "failed to create Zeebe client")
 	defer closeFn()
 
+	s.assertGatewayTopology(err, client)
+	deployProcessResponse := s.deployProcess(err, client)
+	ctx, cancelFn := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancelFn()
+
+	message := retry.DoWithRetry(s.T(), "Try to create Process instance", 10, 1*time.Second, func() (string, error) {
+		_, err = client.NewCreateInstanceCommand().ProcessDefinitionKey(deployProcessResponse.Processes[0].ProcessDefinitionKey).Send(ctx)
+		return "Process instance created.", err
+	})
+	s.T().Logf(message)
+}
+
+func (s *integrationTest) queryProcessDefinitionsFromOperate() (*bytes.Buffer, error) {
+	operateServiceName := fmt.Sprintf("%s-operate", s.release)
+	endpoint, closeFn := s.createPortForwardedHttpClient(operateServiceName)
+	defer closeFn()
+
+	jar, err := cookiejar.New(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient := http.Client{
+		Jar:     jar,
+		Timeout: 30 * time.Second,
+	}
+
+	err = s.loginOnService(endpoint, httpClient)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.queryApi(httpClient, "http://"+endpoint+"/v1/process-definitions/list", bytes.NewBufferString("{}"))
+}
+
+func (s *integrationTest) queryApi(httpClient http.Client, url string, jsonData *bytes.Buffer) (*bytes.Buffer, error) {
+	// curl -i -H "Content-Type: application/json" -XPOST "http://localhost:8080/v1/process-definitions/list" --cookie "ope-session" -d "{}"
+	response, err := httpClient.Post(url, "application/json", jsonData)
+	if err != nil {
+		return nil, err
+	}
+	s.Require().Equal(200, response.StatusCode)
+
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(response.Body)
+	defer response.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+	return buf, nil
+}
+
+func (s *integrationTest) loginOnService(endpoint string, httpClient http.Client) error {
+	// curl --include --request POST --cookie-jar "ope-session" "http://localhost:8080/api/login?username=demo&password=demo"
+	request, err := http.NewRequest("POST", "http://"+endpoint+"/api/login?username=demo&password=demo", nil)
+	if err != nil {
+		return err
+	}
+	request.Close = true
+
+	_, err = httpClient.Do(request)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *integrationTest) awaitCCSMPods() {
+	// await that all ccsm related pods become ready
+	pods := k8s.ListPods(s.T(), s.kubeOptions, v1.ListOptions{LabelSelector: "app=camunda-cloud-self-managed"})
+
+	for _, pod := range pods {
+		k8s.WaitUntilPodAvailable(s.T(), s.kubeOptions, pod.Name, 10, 10*time.Second)
+	}
+}
+
+func (s *integrationTest) awaitElasticPods() {
+	// await that all elastic related pods become ready, otherwise operate and tasklist can't answer requests
+	pods := k8s.ListPods(s.T(), s.kubeOptions, v1.ListOptions{LabelSelector: "app=elasticsearch-master"})
+
+	for _, pod := range pods {
+		k8s.WaitUntilPodAvailable(s.T(), s.kubeOptions, pod.Name, 10, 10*time.Second)
+	}
+}
```

Comment:
just realized that I have some duplication here. Let me fix that in https://github.com/camunda/camunda-cloud-helm/pull/225